tests/rendering/functions-test.ts

```diff
@@ -100,4 +110,29 @@ module('function helpers', function (hooks) {
       'x is 4, and opts.a is 6',
     ]);
   });
+
+  test('the owner is bound as `this` context', async function (assert) {
+    function lookup(
+      this: Owner,
+      identifier: string,
+      options?: object
+    ): unknown {
+      return this.lookup(identifier, options);
+    }
+
+    const testIdentifier = 'test-value:foo';
+    const testValue = 'foo';
+
+    this.owner.register(
+      testIdentifier,
+      { string: testValue },
+      { instantiate: false }
+    );
+
+    this.setProperties({ lookup, testIdentifier });
+
+    await render(hbs`{{get (this.lookup this.testIdentifier) "string"}}`);
```

Comment:
the intent is solid, but ultimately we need the VM to see `property.path` (when used in the invocation position), to do `path.apply(property, args)`

Reply:
Yep! I've touched on that in my most recent comment in the RFC: https://github.com/emberjs/rfcs/pull/756#issuecomment-1210790596

I also suggested not using the `owner`, but the invoking template context. I'd be very interested in your opinion on that.

Reply:
yeah, that would make sense to me.

the owner shouldn't get any special treatment with functions, but your functions should have whatever `this` you expect them to have.

so if your function needs the owner, you `getOwner(this)` in it.
but if your function is not defined in a place where `this` has an owner`, you won't get an owner.


```js
function outerFunction() {
  getOwner(this); // no owner
}

class extends Component {
  // can access owner
  myFunction = () => {
    let owner = getOwner(this);
  }
  
  boundOuterFunction = outerFunction.bind(this); // has owner
}
```